pkg/dns/treecache.go

```diff
@@ -213,13 +181,15 @@ func (cache *TreeCache) GetEntries(recursive bool, path ...string) []interface{}
 	return retval[0]
 }
 
-func (cache *TreeCache) DeletePath(path ...string) bool {
+func (cache *TreeCache) deletePath(path ...string) bool {
 	if len(path) == 0 {
 		return false
 	}
-	cache.m.Lock()
-	defer cache.m.Unlock()
-	if parentNode := cache.GetSubCache(path[:len(path)-1]...); parentNode != nil {
+	if cache.rwLock != nil {
```

Comment:
I'd rather do the locking outside of this function and document that this assumes it is called with the lock held.